specification/search/data-plane/Azure.Search/preview/2025-08-01-preview/knowledgeagent.json

```diff
@@ -0,0 +1,529 @@
+            "$ref": "#/definitions/KnowledgeAgentMessageContent"
+          }
+        }
+      },
+      "required": [
+        "role",
+        "content"
+      ],
+      "description": "The natural language message style object."
+    },
+    "KnowledgeAgentMessageContent": {
+      "type": "object",
+      "discriminator": "type",
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/KnowledgeAgentMessageContentType",
+          "description": "The type of the message",
+          "x-nullable": false
+        }
+      },
+      "required": [
+        "type"
+      ],
+      "description": "Specifies the type of the message content."
+    },
+    "KnowledgeAgentMessageContentType": {
+      "type": "string",
+      "enum": [
+        "text",
+        "image"
+      ],
+      "x-ms-enum": {
+        "name": "KnowledgeAgentMessageContentType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "text",
+            "name": "Text",
+            "description": "Text message content kind."
+          },
+          {
+            "value": "image",
+            "name": "Image",
+            "description": "Image message content kind."
+          }
+        ]
+      },
+      "description": "The type of message content."
+    },
+    "KnowledgeAgentMessageTextContent": {
+      "type": "object",
+      "x-ms-discriminator-value": "text",
+      "allOf": [
+        {
+          "$ref": "#/definitions/KnowledgeAgentMessageContent"
+        }
+      ],
+      "properties": {
+        "text": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "text"
+      ],
+      "description": "Text message type."
+    },
+    "KnowledgeAgentMessageImageContent": {
+      "type": "object",
+      "x-ms-discriminator-value": "image",
+      "allOf": [
+        {
+          "$ref": "#/definitions/KnowledgeAgentMessageContent"
+        }
+      ],
+      "properties": {
+        "image": {
+          "type": "object",
+          "properties": {
+            "url": {
+              "type": "string",
+              "description": "The url of the image."
+            }
+          },
+          "required": [
+            "url"
+          ]
+        }
+      },
+      "required": [
+        "image"
+      ],
+      "description": "Text message type."
+    },
+    "KnowledgeAgentActivityRecord": {
+      "type": "object",
+      "discriminator": "type",
+      "properties": {
+        "id": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The ID of the activity record."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the activity record."
+        }
+      },
+      "required": [
+        "id",
+        "type"
+      ],
+      "description": "Base type for activity records."
+    },
+    "KnowledgeAgentSearchActivityRecord": {
+      "type": "object",
+      "x-ms-discriminator-value": "searchQuery",
```

Comment:
This was `AzureSearchQuery` in 2025-05-01-Preview, was that incorrect in that version or is the value returned being changed for 2025-08-01-Preview?

Reply:
It's being changed for 2025-08-01-Preview. We're normalizing the discriminators to match the KS discriminators and be camelCased rather than PascalCased